apps/console/app/routes/_shell.agent.$agentSlug.branch.$branchSlug.config/route.tsx

```
@@ -0,0 +1,206 @@
+import { BranchModelForm } from "./form";
+import { api } from "~console/lib/data";
+import { RailSymbol } from "lucide-react";
+
+import type { Route } from "./+types/route";
+
+const getModelDisplayName = (modelName: string): string => {
+  if (modelName === "custom") {
+    return "Custom Model";
+  }
+  
+  const model = supportedModels.find((m) => m.name === modelName);
+  return model?.displayName || modelName;
+};
+
+const BranchConfigSchema = object({
+  alias: message(pipe(string(), trim(), nonEmpty()), "Please enter a model alias"),
+  modelType: message(pipe(string(), trim(), nonEmpty()), "Please select a model type"),
+});
+
+export async function clientAction({ request, params }: Route.ClientActionArgs) {
+  const formData = await request.formData();
+  const action = String(formData.get("intent") || formData.get("_action") || "");
+
+  let alias = "";
+  let modelType = "";
+
+  if (action === "remove") {
+    alias = String(formData.get("alias") ?? "");
+    if (!alias.trim()) {
+      return { error: "Alias is required to remove a model" };
+    }
+  } else {
+    const submission = parseWithValibot(formData, { schema: BranchConfigSchema });
+    if (submission.status !== "success") {
+      return { lastResult: submission.reply() };
+    }
+    alias = String(submission.payload.alias ?? "");
+    modelType = String(submission.payload.modelType ?? "");
+
+    if (!supportedModels.some((m) => m.name === modelType)) {
+      return { error: "Selected model type is not supported in this deployment" };
+    }
+  }
+
+  try {
+    // First, fetch the current branch data using Eden treaty client
+    const { data: currentBranch, error: getError } = await api.agents({ agentSlug: params.agentSlug! })
+      .branches({ branchSlug: params.branchSlug! })
+      .get();
+    
+    if (getError) {
+      return { error: getError.value?.toString() || "Failed to fetch branch data" };
+    }
+    const currentModels = currentBranch.models || [];
```

Stop providing empty fallback values, which will hide programming errors.